### ISSUE TITLE
Error building runtime on Windows

### ISSUE THREAD

**Issue by ahmelsayed** [clux/kube-rs#235]
Building `>= 0.32.0` results in

```
   Compiling kube v0.33.0
error[E0432]: unresolved imports `signal::unix`, `signal::unix::SignalKind`
  --> C:\Users\ahmel\.cargo\registry\src\github.com-1ecc6299db9ec823\kube-0.33.0\src\runtime\reflector.rs:62:25
   |
62 |             use signal::unix::{signal, SignalKind}; // TODO: conditional compile
   |                         ^^^^           ^^^^^^^^^^
   |                         |
   |                         could not find `unix` in `signal`

error[E0433]: failed to resolve: use of undeclared type or module `SignalKind`
  --> C:\Users\ahmel\.cargo\registry\src\github.com-1ecc6299db9ec823\kube-0.33.0\src\runtime\reflector.rs:63:38
   |
63 |             let mut sigterm = signal(SignalKind::terminate()).unwrap();
   |                                      ^^^^^^^^^^ use of undeclared type or module `SignalKind`

error[E0659]: `signal` is ambiguous (name vs any other name during import resolution)
  --> C:\Users\ahmel\.cargo\registry\src\github.com-1ecc6299db9ec823\kube-0.33.0\src\runtime\reflector.rs:62:17
   |
62 |             use signal::unix::{signal, SignalKind}; // TODO: conditional compile
   |                 ^^^^^^ ambiguous name
   |
note: `signal` could refer to the unresolved item imported here
  --> C:\Users\ahmel\.cargo\registry\src\github.com-1ecc6299db9ec823\kube-0.33.0\src\runtime\reflector.rs:62:32
   |
62 |             use signal::unix::{signal, SignalKind}; // TODO: conditional compile
   |                                ^^^^^^
note: `signal` could also refer to the module imported here
  --> C:\Users\ahmel\.cargo\registry\src\github.com-1ecc6299db9ec823\kube-0.33.0\src\runtime\reflector.rs:8:14
   |
8  |     signal::{self, ctrl_c},
   |              ^^^^
   = help: use `self::signal` to refer to this module unambiguously

error: aborting due to 3 previous errors
``` 

There is a TODO there for conditional compilation, which seems to be the issue. Just thought I'd open this for tracking 
https://github.com/clux/kube-rs/blob/c5748d3a932feb04631c82f2538e90e61694913f/kube/src/runtime/reflector.rs#L62

### Current workaround:

Downgrade to `0.31.0`

**Comment by clux** [comment]
Yeah, the runtime currently requires a unix signal handler because it needs to intercept SIGTERM in a cluster environment. Conditional compilation would help this (but it's kind of nested inside the futures there), so will probably make the `runtime` a non-default compile time feature for now.

**Comment by clux** [comment]
This was fixed in the referenced pr - forgot to close.

**Comment by clux** [comment]
Released in 0.35.0
